test/DeathTest.cpp

```diff
@@ -144,7 +147,7 @@ TEST(DeathTest, ThreadSafeTest) {
       Death::Instance().RegisterDeathEvent(&DeathTest::RaceTest, "race");
    };
    std::vector<std::future<void>> waitingPromises;
-   for (int i = 0; i < 10000; i++) {
+   for (int i = 0; i < 10; i++) {
```

Comment:
If 10,000 is too excessive, I think 10 is the opposite end of the spectrum (not excessive enough). Can we find a middle ground here with like 100+? I feel like a test with 10 threads is not really stressing the system enough to make the claim "if it works with 10, it will work with 1,000".